src/lib.rs

```diff
@@ -1346,16 +1346,26 @@ pub trait UnwrapThrowExt<T>: Sized {
     /// Unwrap this container's `T` value, or throw an error to JS with the
     /// given message if the `T` value is unavailable (e.g. an `Option<T>` is
     /// `None`).
-    #[cfg_attr(debug_assertions, track_caller)]
+    #[cfg_attr(any(debug_assertions, wasm_bindgen_unwrap_throw_debug), track_caller)]
     fn expect_throw(self, message: &str) -> T;
 }
 
 impl<T> UnwrapThrowExt<T> for Option<T> {
-    #[cfg_attr(debug_assertions, track_caller)]
+    #[cfg(wasm_bindgen_unwrap_throw_debug)]
+    #[track_caller]
+    fn unwrap_throw(self) -> T {
+        self.unwrap()
+    }
```

Comment:
I'm questioning if `wasm-bindgen` should do something here, `unwrap_throw()` should explicitly throw, if it panics it would be doing something different then advertised.

I do agree that the lack of module poisoning is quite problematic, unfortunately I don't know a "quick" fix for it apart from what you have been doing.

I have a couple of suggestions:
- Maybe we can figure out a way to fix `wasm-bindgen-futures`:
  - Now that unwinding is actually supported in Rust Wasm we may want to think about the implementation of `wasm-bindgen-futures` more carefully.
  - If we can't find a good solution, I'm happy to add a `cfg` flag there to panic instead of throwing.
- Other libraries might do whats suggested here, e.g. switch to panicking behind a `cfg` flag.
- Module poisoning needs a solution, I'm happy to look at a PR here in `wasm-bindgen`, even if it should probably be solved in Rustc instead.

Please let me know what you think, I'm sympathetic to your goals but I would like to find a better solution.